### PR TITLE
Fix asymmetric join crash when stream side is empty

### PR DESCRIPTION
**Pull request by jlowe**
Fixes a bug in GpuShuffledAsymmetricHashJoinExec where the code assumed the stream side was not empty, but this is not necessarily the case.  The code now explicitly checks for an empty stream-side iterator and avoids trying to fetch a stream-side batch in that case.

### CODE REVIEW

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffledSizedHashJoinExec.scala
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuShuffledSizedHashJoinExec.scala:1]: The diff is easier to read with whitespace changes hidden.

**Comment by jlowe** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

is it hard to add a test?

**Comment by jlowe** [comment]
> is it hard to add a test?

QA has a test for this, but it's cluster shape specific, took me a while to get it to repro, and it's a little large (a million rows on one side of the join) and takes a bit to run.  I tried quite a while to shrink the test down to something reasonably sized, but each time it would either not reproduce at all or Spark would make the query degenerate.  Given QA has a test for this, I felt like it was covered at least there.

**Pull request merged**